File: 00-每日一题/20200218_88.cpp
//
// Created by zanbo on 2020/2/18.
//
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        for(int i=0;i<n;i++)
        {
            nums1[m+i] = nums2[i];
        }
        //对nums1排序 - 使用快排
        partition(nums1,0,m+n-1);
    }
    void partition(vector<int>& nums,int l,int r)
    {
        if(l>=r) return;
        int i=l;
        int key=nums[l];
        for(int j=l+1;j<=r;j++)
        {
            if(nums[j]<key)
                swap(nums[++i],nums[j]);
        }
        swap(nums[l],nums[i]);

        partition(nums,l,i-1);
        partition(nums,i+1,r);
    }

};

File: 00-每日一题/20200218_88.py
class Solution:
    def merge(self, nums1, m, nums2, n):

        m_j = m - 1
        n_j = n - 1

        for i in range(n + m - 1, -1, -1):

            if m_j < 0:
                nums1[i] = nums2[n_j]
                n_j -= 1
            elif n_j < 0:
                nums1[i] = nums1[m_j]
                m_j -= 1
            elif nums1[m_j] > nums2[n_j]:
                nums1[i] = nums1[m_j]
                m_j -= 1
            else:
                nums1[i] = nums2[n_j]
                n_j -= 1
        return nums1

if __name__ == '__main__':
    obj = Solution()
    while True:
        nums1_str = input().strip().split()
        nums1 = list(map(int, nums1_str))
        m = 0
        for nums in nums1:
            if nums != 0:
                m += 1

        nums2_str = input().strip().split()
        nums2 = list(map(int, nums2_str))
        n = len(nums2)

        res = obj.merge(nums1, m, nums2, n)
        print(res)